### DISCUSSION TITLE
Edges / connected nodes being lost?

### DISCUSSION THREAD

**Discussion by 6r1d** [networkx/networkx#5057] [Q&A] [answered]
Hello. There's a [graph](https://6r1d.github.io/CLIP_graph_visualized/index.html) I saved in JSON before.

I'd like to see nodes, connected to other nodes, and for some reason, the iterator is empty.
Is it a bug or do I misunderstand the process? (I'm using NetworkX `2.5.1`).

```python
from networkx.readwrite import json_graph
import requests

url = 'https://6r1d.github.io/CLIP_graph_visualized/graph.json'
parsed_graph = requests.get(url=url).json()

graph = json_graph.node_link_graph(parsed_graph)

for node in graph.nodes(data=True):
    node_number, node_attrs = node
    if 'name' in node_attrs:
        print('Node: "{}"'.format(node_attrs['name']))
        print(list(graph.neighbors(node_number)))
        print()
```

I can access edges **manually**, but I expected neighboors to provide lists of nodes:

```python
>>> print(graph.edges)
[('0', '1'), ('0', '2'), ('0', '3'), ('0', '4'), ('0', '5'), ('0', '6'), ('0', '14'), ...]
```

**Comment by rossbar** [comment, marked as answer]
It looks like the problem is that the `id` in the node data are integers, whereas the `source` and `target` attributes of the "links" are strings. For instance, when you do `graph.nodes(data=True)`, you will see a set of integer nodes that all have attributes, followed by the same set of nodes (except as strings, not integers) with no attributes.

**Reply by rossbar** [reply]
In this particular case, a bit of manual preprocessing of `parsed_graph` should fix the issue, e.g.

```python
for field in parsed_graph["links"]:
    field["source"] = int(field["source"])
    field["target"] = int(field["target"])

G = json_graph.node_link_graph(parsed_graph)
```

**Reply by 6r1d** [reply]
Thanks, it helped! I should check how I got typing broken on export.
